document-generator-core/src/main/java/uk/gov/companieshouse/document/generator/core/controller/DocumentGeneratorController.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.companieshouse.document.generator.core.controller;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.companieshouse.document.generator.core.models.DocumentGeneratorRequest;
+import uk.gov.companieshouse.document.generator.core.models.DocumentGeneratorResponse;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+import javax.validation.Valid;
+
+import static uk.gov.companieshouse.document.generator.core.DocumentGeneratorApplication.APPLICATION_NAME_SPACE;
+
+@RestController
+public class DocumentGeneratorController {
+
+    private static final Logger LOG = LoggerFactory.getLogger(APPLICATION_NAME_SPACE);
+
+    @PostMapping("/document-generate")
+    @ResponseBody
+    public ResponseEntity generateDocument(@Valid @RequestBody DocumentGeneratorRequest documentGeneratorRequest,
+                                           BindingResult result) {
+
+        DocumentGeneratorResponse response = new DocumentGeneratorResponse();
+
+        if (result.hasErrors()) {
+            LOG.error("error in request body");
+            return new ResponseEntity<>(result.getAllErrors(), HttpStatus.BAD_REQUEST);
+        }
+
+        try {
+            response = null;
+        } catch (Exception e) {
```

Comment:
What are we catching here?

Reply:
As stated in the pr, this is a skeleton controller, I know I will catch an exception here when the service is implemented so I have added the Exception to the catch to ensure it is not forgotten.

Reply:
Is it worth adding a basic service interface so that it can be mocked here and test coverage improved?

Reply:
it is the next stage I am doing, is to implement the service. So in the next process test coverage will be improved.

Reply:
In my opinion we shouldn't be try-catching where it's not appropriate. This could be changed either by adding an interface which throws an expected exception - which would enable more complete testing of the controller - or by refactoring the controller to just set the `response` object to `null` without the try-catch. If the latter is chosen, it could be accompanied by a TODO with a relevant SFA number.